src/Cogworks.AzureSearch/Mappers/AzureSearchParametersMapper.cs

```diff
@@ -0,0 +1,78 @@
+﻿using Azure.Search.Documents;
+using Azure.Search.Documents.Models;
+using Cogworks.AzureSearch.Enums;
+using Cogworks.AzureSearch.Extensions;
+using Cogworks.AzureSearch.Models;
+
+namespace Cogworks.AzureSearch.Mappers
+{
+    public static class AzureSearchParametersMapper
+    {
+        internal static SearchOptions Map(AzureSearchParameters azureSearchParameters)
+        {
+            var searchParameters = new SearchOptions
+            {
+                Filter = azureSearchParameters.Filter,
+                HighlightPostTag = azureSearchParameters.HighlightPostTag,
+                HighlightPreTag = azureSearchParameters.HighlightPreTag,
+                IncludeTotalCount = azureSearchParameters.IncludeTotalResultCount,
+                MinimumCoverage = azureSearchParameters.MinimumCoverage,
+                QueryType = azureSearchParameters.QueryType == AzureQueryType.Full
+                    ? SearchQueryType.Full
+                    : SearchQueryType.Simple,
+                ScoringProfile = azureSearchParameters.ScoringProfile,
+                SearchMode = azureSearchParameters.SearchMode == AzureSearchModeType.Any
+                    ? SearchMode.Any
+                    : SearchMode.All,
+                Skip = azureSearchParameters.Skip,
+                Size = azureSearchParameters.Take
+            };
+
+            if (azureSearchParameters.Select.HasAny())
```

Comment:
Could we do these HasAny() and foreach statements as LINQ instead?

Or maybe it won't be as readable? 🤔

Reply:
I think it won't be readable. But we can try it 🤔 Can you serve some example how it would look like ?